wallet/src/main/scala/org/bitcoins/wallet/LockedWallet.scala

```diff
@@ -73,6 +75,43 @@ abstract class LockedWallet extends LockedWalletApi with BitcoinSLogger {
   override def listAddresses(): Future[Vector[AddressDb]] =
     addressDAO.findAll()
 
+  /** Enumerates the public keys in this wallet */
+  private[wallet] def listPubkeys(): Future[Vector[ECPublicKey]] =
+    addressDAO.findAllPubkeys().map(_.toVector)
+
+  /** Gets the size of the bloom filter for this wallet  */
+  private def getBloomFilterSize(): Future[Int] = {
+    for {
+      pubkeys <- listPubkeys()
+    } yield {
+      // when a public key is inserted into a filter
+      // both the pubkey and the hash of the pubkey
+      // gets inserted
+      pubkeys.length * 2
+    }
+
+  }
+
+  // todo: insert TXIDs? need to track which txids we should
+  // ask for, somehow
+  override def getBloomFilter(): Future[BloomFilter] = {
+    for {
+      pubkeys <- listPubkeys()
+      filterSize <- getBloomFilterSize()
```

Comment:
So this takes care of the case of us _receiving_ money, but doesn't do anything for the case where our utxos are spent on the network. We need to iterate through our `UTXOSpendingInfoTable` and insert the hashes of the script.